de.tudarmstadt.ukp.wikipedia.api/src/main/java/de/tudarmstadt/ukp/wikipedia/api/Category.java

```diff
@@ -134,7 +134,7 @@ private void createCategory(Title title) throws WikiPageNotFoundException {
      * This returns the internal id. Do not confuse this with the pageId.
      * @return Returns the internal id.
      */
-    public long __getId() {
+    long __getId() {
```

Comment:
There is a reason behind this:
I consider the internal, Hibernate/DB specific ID as an internal detail of the DB-access layer. Thus, it should not be exposed to other parts of the application, i.e. no code should rely on accessing the internal database object identifier / primary key. There are other stable data-identifying IDs, such as `pageId`. This idea follows the principle of _information hiding_.

Therefore, I lowered the visibility from `public` and/or `protected` to package private. With all code compiling and all tests passing, I'm confident that nothing will break or suffer from "pain" related to that architectural change.

Reply:
Personally, I'm not very used to using package private things. I have sometimes tried using it to make methods accessible for testing, but very rarely. So I understand the concept of information hiding, but haven't found package private to be very useful for me. Which doesn't mean that it is not useful.

On the other hand, I have seen people often to simple forget to put visibility markers. Unfortunately, Java does not have an explicit marker for package private. I would suggest to put a line comment there if package private is being used intentionally and explain the rationale (e.g. framework-internal use) or maybe even to introduce a custom annotation like `@Internal` just to make the intention explicit and to avoid future developers to think that the modifier has simply been forgotten.

Reply:
Good point. I will leave a comment on the respective pieces of code and push that with the next commit.